hostsfile/src/lib.rs

```diff
@@ -206,9 +210,39 @@ impl HostsBuilder {
         let begin = lines.iter().position(|line| line.trim() == begin_marker);
         let end = lines.iter().position(|line| line.trim() == end_marker);
 
+        let mut lines_to_insert = vec![];
+        if !self.hostname_map.is_empty() {
+            writeln!(lines_to_insert, "{begin_marker}")?;
+            for (ip, hostnames) in &self.hostname_map {
+                if cfg!(windows) {
+                    // windows only allows one hostname per line
+                    for hostname in hostnames {
+                        writeln!(lines_to_insert, "{ip} {hostname}")?;
+                    }
+                } else {
+                    // assume the same format as Unix
+                    writeln!(lines_to_insert, "{} {}", ip, hostnames.join(" "))?;
+                }
+            }
+            writeln!(lines_to_insert, "{end_marker}")?;
+        } else {
+            // There's no need to write to the hosts file if the map is empty
+            return Ok(false);
+        }
+
         let insert = match (begin, end) {
             (Some(begin), Some(end)) => {
-                lines.drain(begin..end + 1);
+                // Add '\n's in between lines
+                let old_section: String = lines
+                    .drain(begin..end + 1)
+                    .fold(String::new(), |acc, line| acc + &line + "\n");
```

Comment:
I would like to use `intersperse`, but that's only available on the nightly channel :/

Reply:
In other parts of the code, we've put something like:

```rust
// TODO: use `intersperse()` when stable, tracking issue: https://github.com/rust-lang/rust/issues/XXX
...
```

So that we can periodically see if new language features allow us to clean up the code eventually :).

Reply:
Since `intersperse` won't be needed anymore, I won't add the comment, but it's a good habit to add those :ok_hand: